packages/react-mongo/react-mongo.ts

```diff
@@ -0,0 +1,120 @@
+import { Meteor } from 'meteor/meteor'
+import { Mongo } from 'meteor/mongo'
+import { Tracker } from 'meteor/tracker'
+import { EJSON } from 'meteor/ejson'
+import { useEffect, useMemo, useReducer, useRef, DependencyList } from 'react'
+
+const fur = (x: number): number => x + 1
+const useForceUpdate = () => useReducer(fur, 0)[1]
+
+type useSubscriptionRefs = {
+  subscription?: Meteor.SubscriptionHandle,
+  computation?: Tracker.Computation,
+  updateOnReady: boolean,
+  isReady: boolean,
+  params: {
+    name?: string,
+    args: any[]
+  }
+}
+
+const useSubscriptionClient = (name?: string, ...args: any[]): Meteor.SubscriptionHandle => {
+  const forceUpdate = useForceUpdate()
+
+  const refs: useSubscriptionRefs = useRef({
+    updateOnReady: false,
+    isReady: false,
+    params: {
+      name,
+      args
+    }
+  }).current
+
+  if (!EJSON.equals(refs.params, { name, args })) {
+    refs.updateOnReady = false
+    refs.isReady = false
+    refs.params = { name, args }
+  }
+
+  useEffect(() => {
+    const computation = Tracker.nonreactive(() => (
+      Tracker.autorun(() => {
+        if (!name) {
+          refs.subscription = null
+          return
+        }
+
+        refs.subscription = Meteor.subscribe(name, ...args)
+
+        const isReady = refs.subscription.ready()
+        if (isReady !== refs.isReady) {
+          refs.isReady = isReady
+          if (refs.updateOnReady) {
+            forceUpdate()
+          }
+        }
+      })
+    ))
+
+    refs.computation = computation
+
+    return () => {
+      computation.stop()
+      refs.subscription = null
+    }
+  }, [name, ...args])
+
+  return {
+    stop () {
+      refs.subscription?.stop()
+    },
+    ready () {
+      // Ready runs synchronously with render, should not create side effects.
+      refs.updateOnReady = true
+      return (refs.subscription && EJSON.equals(refs.params, { name, args }))
```

Comment:
minor perf : I _think_ you could define a single `const paramsEqual = EJSON.equals(refs.params, { name, args })` at the top level of the hook, and use it for both current EJSON.equals calls ?

[edit : or even better - I _think_ unconditionnally returning `refs.isReady` works, as it will be false anyway in the cases tested here]

Reply:
I'll check out both of those options.

Reply:
yeah, just returning `refs.isReady` is the way to go